### DISCUSSION TITLE
lab data in hosp vs icu

### DISCUSSION THREAD

**Discussion by Addison-Weatherhead** [MIT-LCP/mimic-code#1399] [MIMIC-IV]
I'm working with the ICU data, and would like to utilize the lab data for these patients. The [documentation](https://mimic.mit.edu/docs/iv/modules/icu/chartevents/) states that "Some items are duplicated between the labevents and chartevents tables. In cases where there is disagreement between measurements, labevents should be taken as the ground truth". I'm wondering how often these differences occur? In addition, if I'm looking only at labs in labevents.csv (from the hosp module) recorded during ICU stays, will the *number* of recordings be the same? Or will labevents contain extra labs that are simply not recorded in the ICU data?

If the difference is significant enough between lab data stored for icu patients in hops/labevents.csv vs icu/charevents.csv, then I'll probably use labevents. But I've found it difficult to efficiently filter out the lab values recorded during ICU stays, as I have to filter by patient (easy) but then from there filter the labs according to different criteria for each patient (some patients have 1 ICU stay, some have 2, etc. I need to ensure I keep lab measurements recorded in any of the ICU stay windows)

Any advice here? Either on 1) if one can use chartevents' lab values with little worry for noise / loss of information, or 2) how to efficiently extract the lab values from hosp for the icu stays.

Thank you!

**Comment by alistairewj** [comment]
> I'm wondering how often these differences occur?

Not sure, but it could be calculated using some common examples such as potassium. The expected case is that the lab data is updated to reflect an invalid sample, and the charted data is likely not. 

> In addition, if I'm looking only at labs in labevents.csv (from the hosp module) recorded during ICU stays, will the number of recordings be the same? Or will labevents contain extra labs that are simply not recorded in the ICU data?

There will be more labs as (1) they contain measurements outside the ICU and (2) not all types of labs are sent to the ICU system.

I would recommend using the lab data and joining using the patient identifier and a filter on charttime being between the ICU admission start and end. This is relatively easy to do in SQL but it is difficult in Python/pandas. If you want, you can look at some of the first day queries which adopt a similar approach but for the first 24 hours of the ICU stay.

**Reply by Addison-Weatherhead** [reply]
Thank you for the prompt response!

> If you want, you can look at some of the first day queries which adopt a similar approach but for the first 24 hours of the ICU stay.

Sorry, where's this located?

**Reply by Addison-Weatherhead** [reply]
In the mimic-iv-aline-study repo?

**Reply by alistairewj** [reply]
This one is a good example: https://github.com/MIT-LCP/mimic-code/blob/main/mimic-iv/concepts/firstday/first_day_bg.sql

It pulls data from the *labevents* for the first 24 hours of an ICU stay. Should be adaptable to your use case.
